### PR TITLE
Added model Cart

### PR DESCRIPTION
**Pull request by Sundwell**

### CODE REVIEW

File: templates/product/main.html [hunk 1]
```diff
@@ -1,8 +1,22 @@
...
+    <div class="row">
+        {% for product in products %}
+            <div class="col-sm-12 col-md-3 border mb-4 text-center">
+                <img width="100%" src="{{ product.main_photo.url }}" alt="">
+                <h3>{{ product.name }}</h3>
+                <p>Price: {{ product.price }}</p>
+                <p>About: {{ product.short_description }}</p>
+                <a class="btn btn-outline-info btn-block" href="{% url 'product:about' product.id %}">More</a>
+                <button class="btn btn-outline-success btn-block" onclick="addProduct('{{ product.name }}')">Add to cart</button>
+{#                <p>Views: {{ product.views }}</p>#}
```
Comment (VitaliiFisenko) [inline templates/product/main.html]: Remove comments

File: templates/product/main.html [hunk 2]
```diff
@@ -1,8 +1,22 @@
...
+            </div>
+        {% empty %}
```
Comment (VitaliiFisenko) [inline templates/product/main.html]: Empty is redundant here

File: templates/product/about.html [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+                    <span class="carousel-control-next-icon" aria-hidden="true"></span>
+                    <span class="sr-only">Next</span>
+                </a>
+            </div>
+            {#            <a href="#" role="button" class="btn btn-outline-success w-50 btn-block btn-lg mx-auto mt-2" onclick="addProduct({{ product.name }})">Buy</a>#}
+            <button class="btn btn-outline-success w-50 btn-block btn-lg mx-auto mt-2"
+                    onclick="addProduct('{{ product.name }}')">Buy
+            </button>
+        </div>
+        {#    endslider  #}
```
Comment (VitaliiFisenko) [inline templates/product/about.html]: Comments(((

File: templates/product/about.html [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+                <a class="carousel-control-prev" href="#carouselExampleControls" role="button" data-slide="prev">
+                    <span class="carousel-control-prev-icon" aria-hidden="true"></span>
+                    <span class="sr-only">Previous</span>
+                </a>
+                <a class="carousel-control-next" href="#carouselExampleControls" role="button" data-slide="next">
+                    <span class="carousel-control-next-icon" aria-hidden="true"></span>
+                    <span class="sr-only">Next</span>
+                </a>
+            </div>
+            {#            <a href="#" role="button" class="btn btn-outline-success w-50 btn-block btn-lg mx-auto mt-2" onclick="addProduct({{ product.name }})">Buy</a>#}
```
Comment (VitaliiFisenko) [inline templates/product/about.html]: Here they are one more time

File: templates/product/about.html [hunk 5]
```diff
@@ -0,0 +1,57 @@
+{% extends 'base.html' %}
+
+{% block content %}
+    <h1 class="border border-secondary mb-4 mt-4 text-center text-white bg-dark rounded">{{ product.name }}</h1>
+    <div class="row">
+        {#    slider   #}
```
Comment (VitaliiFisenko) [inline templates/product/about.html]: Once again

File: templates/cart/cart.html [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+                </button>
+                <span class="float-right badge badge-secondary mt-4">Sum: {{ cart.cost }}</span>
+            </div>
+        {% else %}
+            <h3 class="text-center">Your cart is empty.</h3>
+        {% endif %}
+    </div>
+{% endblock %}
+{% block extrajs %}
+    <script>
```
Comment (VitaliiFisenko) [inline templates/cart/cart.html:41]: Use separate static folder for static files

File: product/views.py [hunk 7]
```diff
@@ -1,20 +1,39 @@
...
 class ProductAllView(ListView):
-    pass
+    model = Product
+    template_name = 'product/all.html'
+    context_object_name = 'products'
+    paginate_by = 4
+
+    def get_queryset(self, **kwargs):
+        queryset = super().get_queryset()
+        if self.kwargs.get('slug', False) != 'all':
```
Comment (VitaliiFisenko) [inline product/views.py]: use kwargs.get(key, None) instead False. I'll explain why

File: product/models.py [hunk 8]
```diff
@@ -37,6 +37,32 @@ class Product(TimeStampedModel, models.Model):
...
+    )
+    proteins = models.PositiveIntegerField(
+        default=0,
+        null=True,
+    )
+    fats = models.PositiveIntegerField(
+        default=0,
+        null=True,
+    )
+    # Additional
```
Comment (VitaliiFisenko) [inline product/models.py:54]: Comment

File: product/models.py [hunk 9]
```diff
@@ -37,6 +37,32 @@ class Product(TimeStampedModel, models.Model):
         default=0,
         validators=[MinValueValidator(0, 'Wrong price')]
     )
+    # Composition
+    weight = models.DecimalField(
```
Comment (VitaliiFisenko) [inline product/models.py]: Product properties can be one JsonField instead of 5 Integer

File: cart/views.py [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+class ViewCart(ListView):
+    model = Cart
+    template_name = 'cart/cart.html'
+    context_object_name = 'cart'
+
+    def get_queryset(self):
+        return Cart.objects.get(user__id=self.request.user.id)
+
+
+class AddProduct(View):
```
Comment (VitaliiFisenko) [inline cart/views.py:19]: GET not for saving!!!!!! https://www.w3schools.com/tags/ref_httpmethods.asp start with this article

File: cart/templatetags/custom_cart_filters.py [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+    try:
+        return Product.objects.get(name=name).main_photo.url
+    except (AttributeError, Product.DoesNotExist, Product.MultipleObjectsReturned):
+        return None
+
+
+@register.filter
+def get_product_price(name):
+    try:
+        return Product.objects.get(name=name).price
```
Comment (VitaliiFisenko) [inline cart/templatetags/custom_cart_filters.py]: It's not logical. Let's discuss it on Mondey

File: product/models.py (hunk 8 above)
Comment (Sundwell) [reply product/models.py:54]: I think comment may be here

File: product/views.py (hunk 7 above)
Comment (Sundwell) [reply product/views.py]: Done

**Pull request merged**